darshan-util/pydarshan/darshan/lib/accum.py

```diff
@@ -0,0 +1,116 @@
+    >>> log_get_bytes_bandwidth(log_path, "POSIX")
+    I/O performance estimate (at the POSIX layer): transferred 101785.8 MiB at 164.99 MiB/s
+
+    >>> log_get_bytes_bandwidth(log_path, "MPI-IO")
+    I/O performance estimate (at the MPI-IO layer): transferred 126326.8 MiB at 101.58 MiB/s
+    """
+    # get total bytes (in MiB) and bandwidth (in MiB/s) for
+    # a given module -- this information was commonly reported
+    # in the old perl-based summary reports
+    darshan_derived_metrics = log_get_derived_metrics(log_path=log_path,
+                                                      mod_name=mod_name)
+    total_mib = darshan_derived_metrics.total_bytes / 2 ** 20
+    total_bw = darshan_derived_metrics.agg_perf_by_slowest
+    ret_str = f"I/O performance estimate (at the {mod_name} layer): transferred {total_mib:.1f} MiB at {total_bw:.2f} MiB/s"
+    return ret_str
+
+
+def log_file_count_summary_table(log_path: str,
+                                 module: str):
+    # the darshan_file_category enum is not really
+    # exposed in CFFI/Python layer, so we effectively
+    # re-export the content indices we need here
+    # so that we can properly index the C-level data
+    darshan_file_category = {"total opened":0,
+                             "read-only files":1,
+                             "write-only files":2,
+                             "read/write files":3}
+    df = pd.DataFrame.from_dict(darshan_file_category, orient="index")
+    df.rename(columns={0:"index"}, inplace=True)
+    df.index.rename('type', inplace=True)
+    df["number of files"] = np.zeros(4, dtype=int)
+    df["avg. size"] = np.zeros(4, dtype=str)
+    df["max size"] = np.zeros(4, dtype=str)
+
+    darshan_derived_metrics = log_get_derived_metrics(log_path=log_path,
+                                                      mod_name=module)
+    for cat_name, index in darshan_file_category.items():
+        cat_counters = darshan_derived_metrics.category_counters[index]
+        num_files = int(cat_counters.count)
+        if num_files == 0:
+            max_size = "0"
+            avg_size = "0"
+        else:
+            max_size, binary_units = humanize.naturalsize(cat_counters.max_offset_bytes + 1,
+                                                          binary=True).split()
+            # some rounding work to match the old Perl
+            # report table style...
+            max_size = math.ceil(float(max_size)) # type: ignore
+            if max_size == 0:
+                max_size = "0"
+            else:
+                max_size = f"{max_size} {binary_units}"
+            # NOTE: internal formula based on discussion with Phil Carns
+            avg_size, binary_units = humanize.naturalsize((cat_counters.total_max_offset_bytes + num_files) / num_files,
+                                                           binary=True).split()
+            avg_size = math.ceil(float(avg_size)) # type: ignore
```

Comment:
Same for rounding here.

Reply:
done in [aff9aaa](https://github.com/darshan-hpc/darshan/pull/868/commits/aff9aaa071e578fc129d6779bd53165abca1e3fe)